controllers/scopeinstance_controller.go

```diff
@@ -88,12 +89,33 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			return ctrl.Result{}, err
 		}
 
+		cErr := r.updateScopeInstanceCondition(ctx, in, metav1.Condition{
+			Type:               operatorsv1.TypeScoped,
+			Status:             metav1.ConditionFalse,
+			ObservedGeneration: in.Generation,
+			Reason:             operatorsv1.ReasonScopeTemplateNotFound,
+			Message:            fmt.Sprintf("getting ScopeTemplate %q: %s", in.Spec.ScopeTemplateName, err),
+		})
+		if cErr != nil {
+			return ctrl.Result{Requeue: true}, cErr
```

Comment:
No need to `Requeue: true` with a non-nil error.

Also, we've encountered some weirdness in rukpak because our code made updates multiple times in the same reconcile call, which can cause problems with update conflicts, extra reconciles, too much chatter with the apiserver, and it generally makes it harder to reason about stuff.

I'd suggest doing something like what I did here: https://github.com/joelanford/rukpak/blob/1f40e3c07456da15ba8354089ef742fc1859977c/internal/provisioner/plain/controllers/bundle_controller.go#L73-L103. Note that in little r `reconcile` there are no actual update calls. We just mutate the passed in object, and then this function checks to see if the reconciled object differs from the existing object in order to decide whether to issue update calls.

Reply:
@joelanford Oh okay, this makes sense to me. I am happy to get rid of the `Requeue: true` with the non-nil error, but I feel like it may be good to make a follow up issue to tackle making any major changes to the reconciliation loop. WDYT?

Reply:
Yep, sgtm.